src/test/blockmanager_tests.cpp

```diff
@@ -82,4 +83,122 @@ BOOST_FIXTURE_TEST_CASE(blockmanager_scan_unlink_already_pruned_files, TestChain
     BOOST_CHECK(!AutoFile(OpenBlockFile(new_pos, true)).IsNull());
 }
 
+struct FastPruneTestingSetup : public BasicTestingSetup {
+    FastPruneTestingSetup()
+        : BasicTestingSetup{CBaseChainParams::MAIN, /*extra_args=*/{"-fastprune=1"}} {}
+};
+
+BOOST_FIXTURE_TEST_CASE(blockmanager_flush_block_file, FastPruneTestingSetup)
+{
+    const auto params{CreateChainParams(ArgsManager{}, CBaseChainParams::MAIN)};
+    BlockManager blockman{{}};
+    CChain chain{};
+
+    // Ensure (by delaying execution of certain steps)
+    // that if a file is modified its timestamp will be updated.
+#ifdef WIN32
+    // FAT filesystem has write time resolution of 2 seconds
+    constexpr std::chrono::milliseconds pause(2000);
+#else
+    // EXT3 and EXT4 have nanosecond resolution
+    constexpr std::chrono::milliseconds pause(1);
+#endif
+
+    // Test blocks with no transactions, not even a coinbase
+    CBlock block1;
+    block1.nVersion = 1;
+    CBlock block2;
+    block2.nVersion = 2;
+    CBlock block3;
+    block3.nVersion = 3;
+
+    // They are 80 bytes header + 1 byte 0x00 for vtx length
+    constexpr int TEST_BLOCK_SIZE{81};
+
+    // Blockstore is empty
+    BOOST_CHECK_EQUAL(blockman.CalculateCurrentUsage(), 0);
+
+    // Write the first block; dbp=nullptr means this block doesn't already have a disk
+    // location, so allocate a free location and write it there.
+    FlatFilePos pos1{blockman.SaveBlockToDisk(block1, /*nHeight=*/1, chain, *params, /*dbp=*/nullptr)};
+
+    // Write second block
+    FlatFilePos pos2 = blockman.SaveBlockToDisk(block2, 2, chain, *params, nullptr);
+
+    // Two blocks in the file
+    BOOST_CHECK_EQUAL(blockman.CalculateCurrentUsage(), (TEST_BLOCK_SIZE + BLOCK_SERIALIZATION_HEADER_SIZE) * 2);
+
+    // First two blocks are written as expected
+    CBlock read_block;
+    node::ReadBlockFromDisk(read_block, pos1, params->GetConsensus());
+    BOOST_CHECK_EQUAL(read_block.nVersion, 1);
+    node::ReadBlockFromDisk(read_block, pos2, params->GetConsensus());
+    BOOST_CHECK_EQUAL(read_block.nVersion, 2);
+
+    // Record last modified time of blk00000.dat
+    const fs::path file_path = node::GetBlockPosFilename(pos2);
+    const fs::file_time_type time1 = fs::last_write_time(file_path);
+
+    std::this_thread::sleep_for(pause);
```

Comment:
can you explain why you're sleeping _after_ reading `last_write_time`? If I understand correctly, based on the other comments in this thread, the sleep is necessary because the file modified time may not be updated immediately after the contents are written. This test does pass on my linux machine (which has a significantly shorter resolution time), yet I feel that we should be sleeping immediately after the potential file write does/does not occur before checking the file time and verifying it? This comment applies to the other uses of `sleep_for` in this test.